### ISSUE TITLE
Send a HTTP Post alert with specific request body template

### ISSUE THREAD

**Issue by slalwani97** [Yelp/elastalert#2380]
I am using the Http Post alert and the receiver of the alert is predefined with some template that means it has a structure/schema and accepts that particular structure as a request body.

So the structure has a nested field something similar to:
```
{
  "receiver": "elastalert",
  "labels": {
    "alertname": "alertname_from_es_field",
    "summary": "static_value",
  }
}
```

I saw that http_post_payload can only have the list of a key-value pair and cannot be a dictionary.

@Qmando Is there a way to send this (mentioned above) kind of request body in Http Post alert?

#### Solution (Just an approach/thought from my side): 

What if we take the http_post_payload (or some new field with a different name) as a dictionary and parse that dictionary, basically look for the keys which are to be replaced from Elasticsearch field value.

For example:
```
http_post_payload:
  receiver: elastalert
  labels:
    alertname: {es_key}
    summary: something is wrong
```

where keys that are written inside the curly braces (or some other format can be taken) will be looked for values in Elasticsearch data i.e match (using lookup_es_key function).

This can be one approach for code: 
```
def replace_es_key_value_in_string(string, match):
    keys_to_be_replaced = re.findall('{(.+?)}', string)
    # remove duplicates from list
    keys_to_be_replaced = list(dict.fromkeys(keys_to_be_replaced))
    for key in keys_to_be_replaced:
        es_value = lookup_es_key(match, key.strip())
        if es_value == None:
           es_value = ''
        string = string.replace('{' + key + '}', es_value)
    return string

def replace_es_key_value_in_dict_util(body, match):
    if isinstance(body, str):
       return replace_es_key_value_in_string(body, match)

    if isinstance(body, dict):
       for key, val in body.items():
           body[key] = replace_es_key_value_in_dict_util(val, match)

    if isinstance(body, list):
       for val in body:
           val = replace_es_key_value_in_dict_util(val, match)

    return body
```

**Comment by wheezymustafa** [comment]
Yeah I think the `http_post_payload` is a good start but it needs to be flexible enough that we can construct the request body in the way that we need.  Right now we either hardcode values using the `http_post_static_payload` or use post_payload with a rigid body that can't be modified.

**Comment by ashishkaransingh** [comment]
Hello,
I need to send alert to ServiceNow ITOM
Is there a way we can create our own json structure?

**Comment by usharik** [comment]
Let me tell you that those improvements for `http_post_payload` are really useful. Hope finally you are going to implement them.
